### PR TITLE
Fix macos & add to CI

### PR DESCRIPTION
**Pull request by SimeonStoykovQC**
Small test fixes to make it run on macos.

# Checklist

- ~Added a `CHANGELOG.rst` entry~ - only tests/CI change

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,13 +21,14 @@ jobs:
...
-    runs-on: ubuntu-latest
+    name: "Linux - unit tests - ${{ matrix.OS }} - Python ${{ matrix.PYTHON_VERSION }}"
+    runs-on: ${{ matrix.OS }}
     env:
       CI: True
     strategy:
       fail-fast: true
       matrix:
+        OS: ["ubuntu-latest", "macos-latest"]
         PYTHON_VERSION: ['3.8', '3.9', '3.10', '3.11']
```
Comment (xhochy) [inline .github/workflows/ci.yml:32]:
```suggestion
      matrix:
        OS: ["ubuntu-latest"]
        PYTHON_VERSION: ['3.8', '3.9', '3.10', '3.11']
        include:
          - OS: macos-latest-xlarge
             PYTHON_VERSION: "3.11"
```

**Comment by SimeonStoykovQC** [comment]
The opset generation [produced different results](https://github.com/Quantco/spox/actions/runs/6947545870/job/18918033952) to what's in main, so I had to rebase on #115 - or is there a better approach?

File: tests/type_inference/test_compress.py [hunk 2]
```diff
@@ -1,21 +1,26 @@
 import pytest
 
-import spox.opset.ai.onnx.v17 as op
+import spox.opset.ai.onnx.v17 as op17
+import spox.opset.ai.onnx.v18 as op18
+import spox.opset.ai.onnx.v19 as op19
+import spox.opset.ai.onnx.v20 as op20
```
Comment (cbourjau) [inline tests/type_inference/test_compress.py:6]: This got in with the automatic rebase. The parametrize fixture is not needed. We should be fine only using opset version 17.

```suggestion
import spox.opset.ai.onnx.v17 as op
```

File: tests/type_inference/test_compress.py [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
 from spox._graph import arguments
 from spox._standard import InferenceError
 from spox._type_system import Tensor
 
 
-def test_compress_inference():
+@pytest.mark.parametrize("op", [op17, op18, op19, op20])
+def test_compress_inference(op):
```
Comment (cbourjau) [inline tests/type_inference/test_compress.py:13]:
```suggestion
def test_compress_inference():
```

File: tests/type_inference/test_compress.py [hunk 4]
```diff
@@ -1,21 +1,26 @@
...
     x, y = arguments(x=Tensor(float, ("N", "M")), y=Tensor(bool, (None,)))
     assert op.compress(x, y).unwrap_tensor() == Tensor(float, (None,))
     assert op.compress(x, y, axis=0).unwrap_tensor() == Tensor(float, (None, "M"))
     assert op.compress(x, y, axis=1).unwrap_tensor() == Tensor(float, ("N", None))
 
 
-def test_compress_inference_checks_bool_cond():
+@pytest.mark.parametrize("op", [op17, op18, op19, op20])
+def test_compress_inference_checks_bool_cond(op):
```
Comment (cbourjau) [inline tests/type_inference/test_compress.py:21]:
```suggestion
def test_compress_inference_checks_bool_cond():
```

**Review by cbourjau — APPROVED** [review]
Thanks! Good to go if the CI passes after applying those suggestions.

**Pull request merged**